src/main/java/com/nhantran/utils/DriverManager.java

```diff
@@ -1,33 +1,68 @@
+import org.openqa.selenium.Platform;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
 import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.edge.EdgeOptions;
 import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
 
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.time.Duration;
 
 public class DriverManager {
 
     public static WebDriver driver;
 
-    public static void setChromeDriver(String url) {
-        WebDriverManager.chromedriver().setup();
-        driver = new ChromeDriver();
-        navigateToTestSite(url);
-    }
-
-    public static void setFireFoxDriver(String url) {
-        WebDriverManager.firefoxdriver().setup();
-        driver = new FirefoxDriver();
-        navigateToTestSite(url);
+    public static void initLocalDriver(String browser) {
+        switch (browser) {
+            case "chrome":
+                WebDriverManager.chromedriver().setup();
+                driver = new ChromeDriver();
+                break;
+            case "firefox":
+                WebDriverManager.firefoxdriver().setup();
+                driver = new FirefoxDriver();
+                break;
+            case "edge":
+                WebDriverManager.edgedriver().setup();
+                driver = new EdgeDriver();
+                break;
+        }
     }
 
-    public static void setEdgeDriver(String url) {
-        WebDriverManager.firefoxdriver().setup();
-        driver = new EdgeDriver();
-        navigateToTestSite(url);
+    public static void initRemoteDriver(String browser) {
+        try {
+            DesiredCapabilities capability = new DesiredCapabilities();
+            capability.setPlatform(Platform.ANY);
+            switch (browser) {
+                case "chrome":
+                    capability.setBrowserName("chrome");
+                    ChromeOptions chromeOptions = new ChromeOptions();
+                    chromeOptions.merge(capability);
+                    break;
+                case "firefox":
+                    capability.setBrowserName("firefox");
+                    FirefoxOptions firefoxOptions = new FirefoxOptions();
+                    firefoxOptions.merge(capability);
+                    break;
+                case "edge":
+                    capability.setBrowserName("MicrosoftEdge");
+                    EdgeOptions edgeOptions = new EdgeOptions();
+                    edgeOptions.merge(capability);
+                    break;
+            }
+            driver = new RemoteWebDriver(new URL(Constants.NODE_URL), capability);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
+        }
     }
 
     public static void navigateToTestSite(String url) {
```

Comment:
put maximize into init browser, open site just call only navigate to only

Reply:
driver.manage().timeouts().pageLoadTimeout(Duration.ofSeconds(30));
  driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(20));

=> put into the init driver

Reply:
create a timeout variable for this